core/lib/types/src/api/mod.rs

```diff
@@ -728,6 +733,22 @@ pub enum BlockStatus {
     Verified,
 }
 
+/// Result tracers need to have a nested result field for compatibility. So we have two different
+/// structs 1 for blocks tracing and one for txs and call tracing
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum CallTracerResultWithNestedResult {
```

Comment:
- Bikeshedding: This name looks overly long.
- Since this type is only used as `Vec<CallTracerResultWithNestedResult>` (in the output of `debug_traceBlock`), maybe it makes sense to put `Vec<_>` *inside* the enum? AFAIU, it would represent domain better (it's impossible that an output mixes variants from both flat and ordinary tracers). I believe it would make conversions easier as well. Also, it would solve the naming problem; you could call the enum something like `CallTracerBlockResult`.

Reply:
I've refactored this part, but I am not sure it makes it better.

Reply:
Obviously subjective, but I do like it better 👍